app/controllers/users_controller.rb

```diff
@@ -35,46 +35,26 @@ def edit
 
   def update
     if (user_params[:new_password] != '') || (user_params[:email] != @user.email)
-      if user_params[:password] === ''
-        @user.update(user_params)
-        flash.now[:error] = "Password is a mandatory field"
-        render :edit
-      elsif @user.authenticated?(params[:user][:password])
-        if @user.update(user_params)
-          if user_params[:new_password] != ''
-            @user.update_attributes(password: user_params[:new_password])
-          end
-          redirect_to edit_user_path(@user), notice: "You have successfully updated your user data."
-        else
-          render :edit
-        end
-      else
-        flash.now[:error] = "Incorrect password"
-        render :edit
-      end
+      update_protected_params
```

Comment:
this update method has 2 main behaviours:
1.- if the user wants to update their email OR password, in which case we'll require their password confirmation
2.- if the user wants to update other fields that don't need validation. 
so why not using a helper method for the first case (which would be the exception one) so it's clear?